cli/tests/test_utils/mock_server.rs

```diff
@@ -0,0 +1,161 @@
+use std::collections::HashSet;
+use std::fs;
+use std::io::Read;
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
+
+use axum::body::Bytes;
+use axum::extract::State;
+use axum::http::StatusCode;
+use axum::response::Response;
+use axum::routing::{any, post, put};
+use axum::{Json, Router};
+use tempfile::tempdir;
+use tokio::net::TcpListener;
+use tokio::spawn;
+use trunk_analytics_cli::types::{
+    BundleUploadLocation, CreateBundleUploadRequest, CreateRepoRequest,
+    GetQuarantineBulkTestStatusRequest, QuarantineConfig,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum RequestPayload {
+    CreateRepo(CreateRepoRequest),
+    CreateBundleUpload(CreateBundleUploadRequest),
+    GetQuarantineBulkTestStatus(GetQuarantineBulkTestStatusRequest),
+    S3Upload(PathBuf),
+}
+
+#[derive(Debug, Default)]
+pub struct MockServerState {
+    pub requests: Mutex<Vec<RequestPayload>>,
+    pub host: String,
+}
+
+pub type SharedMockServerState = Arc<MockServerState>;
+
+/// Mock server spawned in a new thread.
+///
+/// NOTE: must use a multithreaded executor to have the server run while running tests
+#[allow(dead_code)] // TODO: move this to its own crate to get rid of the need for this
```

Comment:
Regarding this `dead_code` stuff, Rust is setup so that everything within the `tests` module works as integration tests rather than reusable code which is why it complains about these functions. The correct way is to make "`test-utils`" crate and reuse that, but figured there's not really enough here to warrant that. I'm up for splitting it out into a new crate if anyone asks though.